crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -109,7 +104,9 @@ impl ProtocolFees {
             .collect::<Vec<_>>();
 
         if self
-            .protocol_fee_exempt_addresses
+            .cow_amms
+            .list_addresses()
+            .await
```

Comment:
This is called inside a loop(iterating over orders). Wouldn't it be better to get the addresses list before entering the loop and to provide it in the `apply` function args?

Reply:
Agree, would be more straight forward to continue passing the whole set of addresses like we did before.

This also shows that it's inconvenient to have the registry use `async` aware synchronization primitives. Now everything has to become async which can complicate code quite a bit. Maybe a worthwhile improvement would be to make the interface sync. We never hold a lock across `await`s so using the `std::sync` primitives would be fine.

Not sure how well this would translate to the future where we actually persist the events in the db. Maybe it still makes sense for those cases to still maintain an in-memory cache to keep using these sync primitives.